# Proyecto: Precios de Laptos

In [1]:
# Módulo PANDAS
import pandas as pd


In [2]:
# Jalar el data frame
datos=pd.read_csv("laptopPrice.csv")
# Ojo con las diagonales cuando tengan que definir otra ruta..
# pd.read_csv("C:/Users/migue/Documents/Diplomado_ciencia_datos/datos/nombre_archivo.csv")

In [3]:
datos

,brand,processor_brand,processor_name,processor_gnrtn,ram_gb,ram_type,ssd,hdd,os,os_bit,graphic_card_gb,weight,warranty,Touchscreen,msoffice,Price,rating,Number of Ratings,Number of Reviews
0,ASUS,Intel,Core i3,10th,4 GB,DDR4,0 GB,1024 GB,Windows,64-bit,0 GB,Casual,No warranty,No,No,34649,2 stars,3,0
1,Lenovo,Intel,Core i3,10th,4 GB,DDR4,0 GB,1024 GB,Windows,64-bit,0 GB,Casual,No warranty,No,No,38999,3 stars,65,5
2,Lenovo,Intel,Core i3,10th,4 GB,DDR4,0 GB,1024 GB,Windows,64-bit,0 GB,Casual,No warranty,No,No,39999,3 stars,8,1
3,ASUS,Intel,Core i5,10th,8 GB,DDR4,512 GB,0 GB,Windows,32-bit,2 GB,Casual,No warranty,No,No,69990,3 stars,0,0
4,ASUS,Intel,Celeron Dual,Not Available,4 GB,DDR4,0 GB,512 GB,Windows,64-bit,0 GB,Casual,No warranty,No,No,26990,3 stars,0,0
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
818,ASUS,AMD,Ryzen 9,Not Available,4 GB,DDR4,1024 GB,0 GB,Windows,64-bit,0 GB,Casual,1 year,No,No,135990,3 stars,0,0
819,ASUS,AMD,Ryzen 9,Not Available,4 GB,DDR4,1024 GB,0 GB,Windows,64-bit,0 GB,Casual,1 year,No,No,144990,3 stars,0,0
820,ASUS,AMD,Ryzen 9,Not Available,4 GB,DDR4,1024 GB,0 GB,Windows,64-bit,4 GB,Casual,1 year,No,No,149990,3 stars,0,0
821,ASUS,AMD,Ryzen 9,Not Available,4 GB,DDR4,1024 GB,0 GB,Windows,64-bit,4 GB,Casual,1 year,No,No,142990,3 stars,0,0


In [4]:
# Para ver que si se cargo el dataframe
datos.head()  # Las primeras cinco lineas del dataframe

,brand,processor_brand,processor_name,processor_gnrtn,ram_gb,ram_type,ssd,hdd,os,os_bit,graphic_card_gb,weight,warranty,Touchscreen,msoffice,Price,rating,Number of Ratings,Number of Reviews
0,ASUS,Intel,Core i3,10th,4 GB,DDR4,0 GB,1024 GB,Windows,64-bit,0 GB,Casual,No warranty,No,No,34649,2 stars,3,0
1,Lenovo,Intel,Core i3,10th,4 GB,DDR4,0 GB,1024 GB,Windows,64-bit,0 GB,Casual,No warranty,No,No,38999,3 stars,65,5
2,Lenovo,Intel,Core i3,10th,4 GB,DDR4,0 GB,1024 GB,Windows,64-bit,0 GB,Casual,No warranty,No,No,39999,3 stars,8,1
3,ASUS,Intel,Core i5,10th,8 GB,DDR4,512 GB,0 GB,Windows,32-bit,2 GB,Casual,No warranty,No,No,69990,3 stars,0,0
4,ASUS,Intel,Celeron Dual,Not Available,4 GB,DDR4,0 GB,512 GB,Windows,64-bit,0 GB,Casual,No warranty,No,No,26990,3 stars,0,0


In [5]:
# Los últimos 5
datos.tail()

,brand,processor_brand,processor_name,processor_gnrtn,ram_gb,ram_type,ssd,hdd,os,os_bit,graphic_card_gb,weight,warranty,Touchscreen,msoffice,Price,rating,Number of Ratings,Number of Reviews
818,ASUS,AMD,Ryzen 9,Not Available,4 GB,DDR4,1024 GB,0 GB,Windows,64-bit,0 GB,Casual,1 year,No,No,135990,3 stars,0,0
819,ASUS,AMD,Ryzen 9,Not Available,4 GB,DDR4,1024 GB,0 GB,Windows,64-bit,0 GB,Casual,1 year,No,No,144990,3 stars,0,0
820,ASUS,AMD,Ryzen 9,Not Available,4 GB,DDR4,1024 GB,0 GB,Windows,64-bit,4 GB,Casual,1 year,No,No,149990,3 stars,0,0
821,ASUS,AMD,Ryzen 9,Not Available,4 GB,DDR4,1024 GB,0 GB,Windows,64-bit,4 GB,Casual,1 year,No,No,142990,3 stars,0,0
822,Lenovo,AMD,Ryzen 5,10th,8 GB,DDR4,512 GB,0 GB,DOS,64-bit,0 GB,ThinNlight,No warranty,No,No,57490,4 stars,18,4


In [6]:
# datos.head(n)  : les muestra las primera n filas del dataframe datos
datos.head(10)

,brand,processor_brand,processor_name,processor_gnrtn,ram_gb,ram_type,ssd,hdd,os,os_bit,graphic_card_gb,weight,warranty,Touchscreen,msoffice,Price,rating,Number of Ratings,Number of Reviews
0,ASUS,Intel,Core i3,10th,4 GB,DDR4,0 GB,1024 GB,Windows,64-bit,0 GB,Casual,No warranty,No,No,34649,2 stars,3,0
1,Lenovo,Intel,Core i3,10th,4 GB,DDR4,0 GB,1024 GB,Windows,64-bit,0 GB,Casual,No warranty,No,No,38999,3 stars,65,5
2,Lenovo,Intel,Core i3,10th,4 GB,DDR4,0 GB,1024 GB,Windows,64-bit,0 GB,Casual,No warranty,No,No,39999,3 stars,8,1
3,ASUS,Intel,Core i5,10th,8 GB,DDR4,512 GB,0 GB,Windows,32-bit,2 GB,Casual,No warranty,No,No,69990,3 stars,0,0
4,ASUS,Intel,Celeron Dual,Not Available,4 GB,DDR4,0 GB,512 GB,Windows,64-bit,0 GB,Casual,No warranty,No,No,26990,3 stars,0,0
5,ASUS,Intel,Celeron Dual,Not Available,4 GB,DDR4,0 GB,512 GB,Windows,64-bit,0 GB,Casual,No warranty,No,No,22990,3 stars,0,0
6,ASUS,Intel,Celeron Dual,Not Available,4 GB,DDR4,0 GB,512 GB,Windows,32-bit,0 GB,Casual,No warranty,No,No,21990,3 stars,31,3
7,ASUS,Intel,Core i5,10th,8 GB,DDR4,0 GB,1024 GB,Windows,32-bit,2 GB,Casual,No warranty,No,No,58799,3 stars,0,0
8,Lenovo,Intel,Core i5,10th,4 GB,DDR4,0 GB,1024 GB,Windows,32-bit,0 GB,Casual,No warranty,No,No,49999,3 stars,0,0
9,acer,AMD,Ryzen 5,10th,4 GB,DDR4,0 GB,512 GB,Windows,32-bit,4 GB,Casual,No warranty,No,No,59990,4 stars,1946,240


In [7]:
# datos.tail(n)  : les muestra las ultimas n filas del dataframe datos
datos.tail(10)

,brand,processor_brand,processor_name,processor_gnrtn,ram_gb,ram_type,ssd,hdd,os,os_bit,graphic_card_gb,weight,warranty,Touchscreen,msoffice,Price,rating,Number of Ratings,Number of Reviews
813,ASUS,AMD,Ryzen 9,Not Available,4 GB,DDR4,1024 GB,0 GB,Windows,64-bit,4 GB,Casual,1 year,No,No,162990,3 stars,0,0
814,ASUS,AMD,Ryzen 9,Not Available,4 GB,DDR4,1024 GB,0 GB,Windows,64-bit,0 GB,Casual,1 year,No,No,130500,3 stars,0,0
815,ASUS,AMD,Ryzen 9,Not Available,4 GB,DDR4,1024 GB,0 GB,Windows,64-bit,4 GB,Casual,No warranty,No,No,116990,3 stars,0,0
816,ASUS,AMD,Ryzen 9,Not Available,4 GB,DDR4,1024 GB,0 GB,Windows,32-bit,6 GB,Casual,No warranty,No,No,138990,3 stars,0,0
817,ASUS,AMD,Ryzen 9,Not Available,4 GB,LPDDR4X,1024 GB,0 GB,Windows,64-bit,4 GB,Casual,1 year,Yes,Yes,234990,3 stars,0,0
818,ASUS,AMD,Ryzen 9,Not Available,4 GB,DDR4,1024 GB,0 GB,Windows,64-bit,0 GB,Casual,1 year,No,No,135990,3 stars,0,0
819,ASUS,AMD,Ryzen 9,Not Available,4 GB,DDR4,1024 GB,0 GB,Windows,64-bit,0 GB,Casual,1 year,No,No,144990,3 stars,0,0
820,ASUS,AMD,Ryzen 9,Not Available,4 GB,DDR4,1024 GB,0 GB,Windows,64-bit,4 GB,Casual,1 year,No,No,149990,3 stars,0,0
821,ASUS,AMD,Ryzen 9,Not Available,4 GB,DDR4,1024 GB,0 GB,Windows,64-bit,4 GB,Casual,1 year,No,No,142990,3 stars,0,0
822,Lenovo,AMD,Ryzen 5,10th,8 GB,DDR4,512 GB,0 GB,DOS,64-bit,0 GB,ThinNlight,No warranty,No,No,57490,4 stars,18,4


# Ayuda con los métodos de los módulos.
1. Para hallar los métodos que están definidos en cada uno de los modulos de Python, se puede hacer los siguiente:

nombre_del_modulo.  acontinuación presionamos tecla de tabular
(o el alias)

2. Para revisar los parámetros de un método en python procedemos como sigue:

nombre_del_modulo.nombre_del_método( acontinuación presionamos las teclas shift+ tab
(o el alias)


In [13]:
# Ejemplos:
#pd.concat()

# Siempre hay que importar el módulo primero... antes de solicitar la ayuda
import os
#os.chdir()



In [14]:
# Cómo mostramos TODAS las entradas del DataFrame 
pd.set_option("display.max_rows",1000)  # Le cambio el parámetro para que me despliegue hasta 1000 renglones
datos

,brand,processor_brand,processor_name,processor_gnrtn,ram_gb,ram_type,ssd,hdd,os,os_bit,graphic_card_gb,weight,warranty,Touchscreen,msoffice,Price,rating,Number of Ratings,Number of Reviews
0,ASUS,Intel,Core i3,10th,4 GB,DDR4,0 GB,1024 GB,Windows,64-bit,0 GB,Casual,No warranty,No,No,34649,2 stars,3,0
1,Lenovo,Intel,Core i3,10th,4 GB,DDR4,0 GB,1024 GB,Windows,64-bit,0 GB,Casual,No warranty,No,No,38999,3 stars,65,5
2,Lenovo,Intel,Core i3,10th,4 GB,DDR4,0 GB,1024 GB,Windows,64-bit,0 GB,Casual,No warranty,No,No,39999,3 stars,8,1
3,ASUS,Intel,Core i5,10th,8 GB,DDR4,512 GB,0 GB,Windows,32-bit,2 GB,Casual,No warranty,No,No,69990,3 stars,0,0
4,ASUS,Intel,Celeron Dual,Not Available,4 GB,DDR4,0 GB,512 GB,Windows,64-bit,0 GB,Casual,No warranty,No,No,26990,3 stars,0,0
5,ASUS,Intel,Celeron Dual,Not Available,4 GB,DDR4,0 GB,512 GB,Windows,64-bit,0 GB,Casual,No warranty,No,No,22990,3 stars,0,0
6,ASUS,Intel,Celeron Dual,Not Available,4 GB,DDR4,0 GB,512 GB,Windows,32-bit,0 GB,Casual,No warranty,No,No,21990,3 stars,31,3
7,ASUS,Intel,Core i5,10th,8 GB,DDR4,0 GB,1024 GB,Windows,32-bit,2 GB,Casual,No warranty,No,No,58799,3 stars,0,0
8,Lenovo,Intel,Core i5,10th,4 GB,DDR4,0 GB,1024 GB,Windows,32-bit,0 GB,Casual,No warranty,No,No,49999,3 stars,0,0
9,acer,AMD,Ryzen 5,10th,4 GB,DDR4,0 GB,512 GB,Windows,32-bit,4 GB,Casual,No warranty,No,No,59990,4 stars,1946,240


# Atributos del DataFrame


In [15]:
# Dimensiones del dataframe: atributo 
datos.shape

(823, 19)

Tenemos 823 laptops y en cada una de ellas hemos medido 19 atributos


In [16]:
# Los atributos son: Las variables que estamos midiendo
# con el atributo columns
datos.columns

Index(['brand', 'processor_brand', 'processor_name', 'processor_gnrtn',
       'ram_gb', 'ram_type', 'ssd', 'hdd', 'os', 'os_bit', 'graphic_card_gb',
       'weight', 'warranty', 'Touchscreen', 'msoffice', 'Price', 'rating',
       'Number of Ratings', 'Number of Reviews'],
      dtype='object')

In [17]:
datos.dtypes

brand                object
processor_brand      object
processor_name       object
processor_gnrtn      object
ram_gb               object
ram_type             object
ssd                  object
hdd                  object
os                   object
os_bit               object
graphic_card_gb      object
weight               object
warranty             object
Touchscreen          object
msoffice             object
Price                 int64
rating               object
Number of Ratings     int64
Number of Reviews     int64
dtype: object

In [18]:
# Método más general para obtener la información del dataframe
datos.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 823 entries, 0 to 822
Data columns (total 19 columns):
 #   Column             Non-Null Count  Dtype 
---  ------             --------------  ----- 
 0   brand              823 non-null    object
 1   processor_brand    823 non-null    object
 2   processor_name     823 non-null    object
 3   processor_gnrtn    823 non-null    object
 4   ram_gb             823 non-null    object
 5   ram_type           823 non-null    object
 6   ssd                823 non-null    object
 7   hdd                823 non-null    object
 8   os                 823 non-null    object
 9   os_bit             823 non-null    object
 10  graphic_card_gb    823 non-null    object
 11  weight             823 non-null    object
 12  warranty           823 non-null    object
 13  Touchscreen        823 non-null    object
 14  msoffice           823 non-null    object
 15  Price              823 non-null    int64 
 16  rating             823 non-null    object
 1

In [19]:
datos.describe()

,Price,Number of Ratings,Number of Reviews
count,823.000000,823.000000,823.000000
mean,76745.177400,315.301337,37.609964
std,45101.790525,1047.382654,121.728017
min,16990.000000,0.000000,0.000000
25%,46095.000000,0.000000,0.000000
50%,64990.000000,17.000000,2.000000
75%,89636.000000,139.500000,18.000000
max,441990.000000,15279.000000,1947.000000


# Funciones sobre DataFrame


In [20]:
len(datos)  # Regresa el número de renglones en el dataframe

823

In [21]:
round(datos,2)  # Redondeo a 2 cifras decimales, cuando aplica para variables numéricas

,brand,processor_brand,processor_name,processor_gnrtn,ram_gb,ram_type,ssd,hdd,os,os_bit,graphic_card_gb,weight,warranty,Touchscreen,msoffice,Price,rating,Number of Ratings,Number of Reviews
0,ASUS,Intel,Core i3,10th,4 GB,DDR4,0 GB,1024 GB,Windows,64-bit,0 GB,Casual,No warranty,No,No,34649,2 stars,3,0
1,Lenovo,Intel,Core i3,10th,4 GB,DDR4,0 GB,1024 GB,Windows,64-bit,0 GB,Casual,No warranty,No,No,38999,3 stars,65,5
2,Lenovo,Intel,Core i3,10th,4 GB,DDR4,0 GB,1024 GB,Windows,64-bit,0 GB,Casual,No warranty,No,No,39999,3 stars,8,1
3,ASUS,Intel,Core i5,10th,8 GB,DDR4,512 GB,0 GB,Windows,32-bit,2 GB,Casual,No warranty,No,No,69990,3 stars,0,0
4,ASUS,Intel,Celeron Dual,Not Available,4 GB,DDR4,0 GB,512 GB,Windows,64-bit,0 GB,Casual,No warranty,No,No,26990,3 stars,0,0
5,ASUS,Intel,Celeron Dual,Not Available,4 GB,DDR4,0 GB,512 GB,Windows,64-bit,0 GB,Casual,No warranty,No,No,22990,3 stars,0,0
6,ASUS,Intel,Celeron Dual,Not Available,4 GB,DDR4,0 GB,512 GB,Windows,32-bit,0 GB,Casual,No warranty,No,No,21990,3 stars,31,3
7,ASUS,Intel,Core i5,10th,8 GB,DDR4,0 GB,1024 GB,Windows,32-bit,2 GB,Casual,No warranty,No,No,58799,3 stars,0,0
8,Lenovo,Intel,Core i5,10th,4 GB,DDR4,0 GB,1024 GB,Windows,32-bit,0 GB,Casual,No warranty,No,No,49999,3 stars,0,0
9,acer,AMD,Ryzen 5,10th,4 GB,DDR4,0 GB,512 GB,Windows,32-bit,4 GB,Casual,No warranty,No,No,59990,4 stars,1946,240


In [23]:
# Qué marcas de Laptop están en tu dataframe?

# Seleccionar una serie/columna de un dataframe
# sintaxis: nombre_dataframe["nombre_serie"]
datos["brand"] # Seleccionamos solo la marca de la laptop
datos["brand"].unique()   # Devuelve los valores únicos de la marca

array(['ASUS', 'Lenovo', 'acer', 'Avita', 'HP', 'DELL', 'MSI', 'APPLE'],
      dtype=object)

In [24]:
# ¿Cuántas computadoras de cada marca están en tu estudio?
datos["brand"].value_counts()   # Devuelve el número de laptops de cada marca

ASUS      249
DELL      151
Lenovo    144
HP        136
acer       51
MSI        50
APPLE      28
Avita      14
Name: brand, dtype: int64

In [25]:
# En porcentajes?
datos["brand"].value_counts(normalize=True)   # Devuelve el porcentaje de laptops de cada marca


ASUS      0.302552
DELL      0.183475
Lenovo    0.174970
HP        0.165249
acer      0.061968
MSI       0.060753
APPLE     0.034022
Avita     0.017011
Name: brand, dtype: float64

In [ ]:
# El 30.25% de las laptops que más se venden son de la marca ASUS.

In [29]:
# Seleccionar una numérica, por ejemplo precio..
datos["Price"]

0       34649
1       38999
2       39999
3       69990
4       26990
5       22990
6       21990
7       58799
8       49999
9       59990
10      93700
11      72990
12      17490
13      22990
14      35990
15      56490
16      65390
17      31999
18      32490
19      31799
20      29890
21      23990
22      32955
23      46200
24      34429
25      37990
26      35990
27      33980
28      39990
29      39490
30      53990
31      38990
32      37990
33      37890
34      38990
35      37990
36      41990
37      45990
38      39990
39      42490
40      36990
41      53990
42      36990
43      38940
44      39490
45      45990
46      39999
47      42990
48      49990
49      41490
50      41990
51      41990
52      38490
53      38990
54      44990
55      36490
56      38970
57      44900
58      38990
59      44427
60      35990
61      40200
62      40990
63      42490
64      23990
65      71999
66      39990
67      41998
68      41990
69      39999
70      39990
71    

In [32]:
# Otra manera de seleccionar una variable es con el punto.
# nombre_dataframe.nombre_serie
datos.Price

0       34649
1       38999
2       39999
3       69990
4       26990
5       22990
6       21990
7       58799
8       49999
9       59990
10      93700
11      72990
12      17490
13      22990
14      35990
15      56490
16      65390
17      31999
18      32490
19      31799
20      29890
21      23990
22      32955
23      46200
24      34429
25      37990
26      35990
27      33980
28      39990
29      39490
30      53990
31      38990
32      37990
33      37890
34      38990
35      37990
36      41990
37      45990
38      39990
39      42490
40      36990
41      53990
42      36990
43      38940
44      39490
45      45990
46      39999
47      42990
48      49990
49      41490
50      41990
51      41990
52      38490
53      38990
54      44990
55      36490
56      38970
57      44900
58      38990
59      44427
60      35990
61      40200
62      40990
63      42490
64      23990
65      71999
66      39990
67      41998
68      41990
69      39999
70      39990
71    

In [35]:
# Precio promedio de las laptops:
datos["Price"].mean()  # Devuelve el precio promedio
datos["Price"].min()   # Devuelve el precio de la computadora más barata
datos["Price"].max()   # Devuelve el precio de la computadora más cara
# Ojo! Solo imprime la última instrucción

441990

In [49]:
print("El precio promedio de las computadoras portátiles es {:.3f} +/- {:.3f} dólares".
      format(datos["Price"].mean(),datos["Price"].std()))

El precio promedio de las computadoras portátiles es 76745.177 +/- 45101.791 dólares


In [46]:
# Opcional: 
precio_promedio=datos["Price"].mean()
print(f"El precio promedio de las computadoras portátiles es {precio_promedio}")

El precio promedio de las computadoras portátiles es 76745.17739975698


In [56]:
# Opcional:
print("El precio promedio de las computadoras es: ", datos["Price"].mean())

El precio promedio de las computadoras es:  76745.17739975698


In [52]:
print("El precio promedio de las computadoras portátiles es {:.2f}".format(datos["Price"].mean()))
print("La computadora portátil más económica tiene un precio de {:.2f}".format(datos["Price"].min()))
print("La computadora portátil más cara tiene un precio de {:.2f}".format(datos["Price"].max()))

El precio promedio de las computadoras portátiles es 76745.18
La computadora portátil más económica tiene un precio de 16990.00
La computadora portátil más cara tiene un precio de 441990.00


In [54]:
datos.Price.std()

45101.790524689466

In [55]:
datos["Price"].sum()  # Devuelve la suma de los datos de la serie

63161281

# Filtrado de un DataFrame

In [62]:
# Solo me interesa una marca. ASUS.
datos[datos["brand"]=="ASUS"]  # Devuelve el dataframe de las laptops ASUS


,brand,processor_brand,processor_name,processor_gnrtn,ram_gb,ram_type,ssd,hdd,os,os_bit,graphic_card_gb,weight,warranty,Touchscreen,msoffice,Price,rating,Number of Ratings,Number of Reviews
0,ASUS,Intel,Core i3,10th,4 GB,DDR4,0 GB,1024 GB,Windows,64-bit,0 GB,Casual,No warranty,No,No,34649,2 stars,3,0
3,ASUS,Intel,Core i5,10th,8 GB,DDR4,512 GB,0 GB,Windows,32-bit,2 GB,Casual,No warranty,No,No,69990,3 stars,0,0
4,ASUS,Intel,Celeron Dual,Not Available,4 GB,DDR4,0 GB,512 GB,Windows,64-bit,0 GB,Casual,No warranty,No,No,26990,3 stars,0,0
5,ASUS,Intel,Celeron Dual,Not Available,4 GB,DDR4,0 GB,512 GB,Windows,64-bit,0 GB,Casual,No warranty,No,No,22990,3 stars,0,0
6,ASUS,Intel,Celeron Dual,Not Available,4 GB,DDR4,0 GB,512 GB,Windows,32-bit,0 GB,Casual,No warranty,No,No,21990,3 stars,31,3
7,ASUS,Intel,Core i5,10th,8 GB,DDR4,0 GB,1024 GB,Windows,32-bit,2 GB,Casual,No warranty,No,No,58799,3 stars,0,0
13,ASUS,Intel,Celeron Dual,Not Available,4 GB,DDR4,0 GB,512 GB,Windows,64-bit,0 GB,Casual,No warranty,No,No,22990,3 stars,0,0
14,ASUS,Intel,Core i3,11th,4 GB,DDR4,256 GB,0 GB,Windows,64-bit,0 GB,Casual,No warranty,No,No,35990,3 stars,0,0
20,ASUS,Intel,Celeron Dual,Not Available,4 GB,DDR4,256 GB,0 GB,Windows,64-bit,0 GB,Casual,No warranty,No,No,29890,4 stars,7,0
21,ASUS,Intel,Celeron Dual,Not Available,4 GB,DDR4,0 GB,1024 GB,Windows,64-bit,0 GB,Casual,1 year,No,No,23990,3 stars,15279,1947


In [63]:
#Si me interesa solo el precio
datos[datos["brand"]=="ASUS"]["Price"]  # Devuelve el precio de las laptops ASUS


0       34649
3       69990
4       26990
5       22990
6       21990
7       58799
13      22990
14      35990
20      29890
21      23990
24      34429
26      35990
27      33980
32      37990
33      37890
37      45990
42      36990
45      45990
52      38490
60      35990
66      39990
68      41990
70      39990
74      48490
81      36990
83      48990
84      23490
86      37990
89      35990
90      42990
91      51990
92      48990
93      42990
98      39990
102     41990
104     44990
107     42990
112     44990
117     51649
119     40990
120     42990
121     44890
122     41490
125     40990
130     32890
136     35990
139     34990
140     38900
148     35590
150     38990
151     33990
153     24990
156     38000
160     41500
163     21490
165     32990
166     38680
167     23990
169     33990
170     20490
174     41890
180     40990
188     18990
189     17990
199    106167
208     48990
213     49800
219     57990
224     63990
228     71990
229     47990
231   

In [65]:
# El precio promedio de venta de las computadoras ASUS
print("Precio promedio de las laptops marca ASUS {:.2f} ".format(datos[datos["brand"]=="ASUS"]["Price"].mean()))

Precio promedio de las laptops marca ASUS 78937.60 


In [66]:
# El precio promedio de venta de las computadoras HP
print("Precio promedio de las laptops marca HP {:.2f} ".format(datos[datos["brand"]=="HP"]["Price"].mean()))

Precio promedio de las laptops marca HP 73640.27 


In [67]:
# El precio promedio de venta de las computadoras DELL
print("Precio promedio de las laptops marca DELL {:.2f} ".format(datos[datos["brand"]=="DELL"]["Price"].mean()))

Precio promedio de las laptops marca DELL 60934.66 


Para simplificar el trabajo no vamos a apoyar de un loop for... para hacer el cálculo
de manera individual por marca.

In [69]:
datos["brand"].unique()   # Marcas en mi dataframe

array(['ASUS', 'Lenovo', 'acer', 'Avita', 'HP', 'DELL', 'MSI', 'APPLE'],
      dtype=object)

In [70]:
marcas=datos["brand"].unique()   # Marcas en mi dataframe

In [71]:
marcas

array(['ASUS', 'Lenovo', 'acer', 'Avita', 'HP', 'DELL', 'MSI', 'APPLE'],
      dtype=object)

In [73]:
for marca in marcas:
    print("El precio promedio de las laptops marca {} es {:.2f} ".format(marca,datos[datos["brand"]==marca]["Price"].mean()))
    

El precio promedio de las laptops marca ASUS es 78937.60 
El precio promedio de las laptops marca Lenovo es 72920.21 
El precio promedio de las laptops marca acer es 72420.04 
El precio promedio de las laptops marca Avita es 65157.43 
El precio promedio de las laptops marca HP es 73640.27 
El precio promedio de las laptops marca DELL es 60934.66 
El precio promedio de las laptops marca MSI es 98713.02 
El precio promedio de las laptops marca APPLE es 151707.86 


In [ ]:
# Repetir este ejercicio de calcular el precio promedio de venta, pero ahora por 
# capacidad de memoria RAM.
# Cuál es el precio de venta de una computadora con 4 GB en ram?
# Cuál es el precio de venta de una computadora con 8 GB en ram?
# Cuál es el precio de venta de una computadora con 16 GB en ram?
# Cuál es el precio de venta de una computadora con 32 GB en ram?

# Operaciones entre columnas de un DataFrame

In [76]:
# Supongamos que el precio de venta es con impuestos... cómo obtengo los precios
# de las laptops antes de impuestos.. suponiendo un impuesto de 16%

datos["Price"]/1.16


0       29869.827586
1       33619.827586
2       34481.896552
3       60336.206897
4       23267.241379
5       19818.965517
6       18956.896552
7       50688.793103
8       43102.586207
9       51715.517241
10      80775.862069
11      62922.413793
12      15077.586207
13      19818.965517
14      31025.862069
15      48698.275862
16      56370.689655
17      27585.344828
18      28008.620690
19      27412.931034
20      25767.241379
21      20681.034483
22      28409.482759
23      39827.586207
24      29680.172414
25      32750.000000
26      31025.862069
27      29293.103448
28      34474.137931
29      34043.103448
30      46543.103448
31      33612.068966
32      32750.000000
33      32663.793103
34      33612.068966
35      32750.000000
36      36198.275862
37      39646.551724
38      34474.137931
39      36629.310345
40      31887.931034
41      46543.103448
42      31887.931034
43      33568.965517
44      34043.103448
45      39646.551724
46      34481.896552
47      37060

In [77]:
# Cómo creamos una nueva serie en el DataFrame
datos["Price before taxes"]=datos["Price"]/1.16


In [78]:
datos.head()

,brand,processor_brand,processor_name,processor_gnrtn,ram_gb,ram_type,ssd,hdd,os,os_bit,graphic_card_gb,weight,warranty,Touchscreen,msoffice,Price,rating,Number of Ratings,Number of Reviews,Price before taxes
0,ASUS,Intel,Core i3,10th,4 GB,DDR4,0 GB,1024 GB,Windows,64-bit,0 GB,Casual,No warranty,No,No,34649,2 stars,3,0,29869.827586
1,Lenovo,Intel,Core i3,10th,4 GB,DDR4,0 GB,1024 GB,Windows,64-bit,0 GB,Casual,No warranty,No,No,38999,3 stars,65,5,33619.827586
2,Lenovo,Intel,Core i3,10th,4 GB,DDR4,0 GB,1024 GB,Windows,64-bit,0 GB,Casual,No warranty,No,No,39999,3 stars,8,1,34481.896552
3,ASUS,Intel,Core i5,10th,8 GB,DDR4,512 GB,0 GB,Windows,32-bit,2 GB,Casual,No warranty,No,No,69990,3 stars,0,0,60336.206897
4,ASUS,Intel,Celeron Dual,Not Available,4 GB,DDR4,0 GB,512 GB,Windows,64-bit,0 GB,Casual,No warranty,No,No,26990,3 stars,0,0,23267.241379


In [79]:
# Cuánto paga la tienda por impuestos en la venta de computadoras
datos["Taxes"]=datos["Price"]-datos["Price before taxes"]
datos.head()


,brand,processor_brand,processor_name,processor_gnrtn,ram_gb,ram_type,ssd,hdd,os,os_bit,...,weight,warranty,Touchscreen,msoffice,Price,rating,Number of Ratings,Number of Reviews,Price before taxes,Taxes
0,ASUS,Intel,Core i3,10th,4 GB,DDR4,0 GB,1024 GB,Windows,64-bit,...,Casual,No warranty,No,No,34649,2 stars,3,0,29869.827586,4779.172414
1,Lenovo,Intel,Core i3,10th,4 GB,DDR4,0 GB,1024 GB,Windows,64-bit,...,Casual,No warranty,No,No,38999,3 stars,65,5,33619.827586,5379.172414
2,Lenovo,Intel,Core i3,10th,4 GB,DDR4,0 GB,1024 GB,Windows,64-bit,...,Casual,No warranty,No,No,39999,3 stars,8,1,34481.896552,5517.103448
3,ASUS,Intel,Core i5,10th,8 GB,DDR4,512 GB,0 GB,Windows,32-bit,...,Casual,No warranty,No,No,69990,3 stars,0,0,60336.206897,9653.793103
4,ASUS,Intel,Celeron Dual,Not Available,4 GB,DDR4,0 GB,512 GB,Windows,64-bit,...,Casual,No warranty,No,No,26990,3 stars,0,0,23267.241379,3722.758621


In [80]:
datos["Taxes"].sum()  # Cantidad total a pagar por impuestos

8711900.827586202

In [81]:
datos["Taxes"].mean() # Impuestos promedio

10585.541710311309

# Ordenar un DataFrame
Sintaxis:

nombre_dataframe.sort_values(by="nombre_serie")


IMPORTANTE: Estos cambios no afectan el dataframe original, solo aplican para desplegar el
resultado en la pantalla. Si se desea que los cambios se apliquen al DataFrame original
se debe emplear el parámetro _inplace_=True

In [82]:
datos.sort_values(by="brand")  # Ordena por marca en orden creciente

,brand,processor_brand,processor_name,processor_gnrtn,ram_gb,ram_type,ssd,hdd,os,os_bit,...,weight,warranty,Touchscreen,msoffice,Price,rating,Number of Ratings,Number of Reviews,Price before taxes,Taxes
609,APPLE,M1,M1,10th,8 GB,DDR4,512 GB,0 GB,Mac,64-bit,...,Casual,1 year,No,No,101990,4 stars,809,91,87922.413793,14067.586207
604,APPLE,M1,M1,10th,16 GB,DDR4,256 GB,0 GB,Mac,64-bit,...,Casual,1 year,No,No,104990,4 stars,45,4,90508.620690,14481.379310
605,APPLE,M1,M1,10th,8 GB,DDR4,256 GB,0 GB,Mac,64-bit,...,Casual,1 year,No,No,85990,4 stars,6725,629,74129.310345,11860.689655
606,APPLE,M1,M1,10th,16 GB,DDR4,512 GB,0 GB,Mac,64-bit,...,Casual,1 year,No,No,149990,4 stars,640,70,129301.724138,20688.275862
607,APPLE,M1,M1,10th,8 GB,DDR4,512 GB,0 GB,Mac,64-bit,...,Casual,1 year,No,No,125990,4 stars,399,45,108612.068966,17377.931034
309,APPLE,Intel,Core i5,10th,8 GB,LPDDR4X,512 GB,0 GB,Mac,64-bit,...,Casual,1 year,No,No,159990,4 stars,32,5,137922.413793,22067.586207
310,APPLE,Intel,Core i5,8th,8 GB,DDR3,512 GB,0 GB,Mac,64-bit,...,Casual,1 year,No,No,172090,4 stars,108,13,148353.448276,23736.551724
608,APPLE,M1,M1,10th,8 GB,DDR4,512 GB,0 GB,Mac,64-bit,...,Casual,1 year,No,No,103990,4 stars,809,91,89646.551724,14343.448276
621,APPLE,M1,M1,10th,32 GB,DDR4,1024 GB,0 GB,Mac,64-bit,...,Casual,1 year,No,No,309990,5 stars,3,1,267232.758621,42757.241379
610,APPLE,M1,M1,10th,8 GB,DDR4,512 GB,0 GB,Mac,64-bit,...,Casual,1 year,No,No,104990,4 stars,6725,629,90508.620690,14481.379310


In [83]:
datos.sort_values(by="Price")  # Ordena por precio en sentido creciente 

,brand,processor_brand,processor_name,processor_gnrtn,ram_gb,ram_type,ssd,hdd,os,os_bit,...,weight,warranty,Touchscreen,msoffice,Price,rating,Number of Ratings,Number of Reviews,Price before taxes,Taxes
194,Lenovo,Intel,Celeron Dual,Not Available,4 GB,LPDDR4,0 GB,512 GB,Windows,32-bit,...,Casual,No warranty,No,No,16990,3 stars,414,54,14646.551724,2343.448276
12,Avita,Intel,Celeron Dual,Not Available,4 GB,DDR4,0 GB,512 GB,Windows,64-bit,...,Casual,No warranty,No,No,17490,3 stars,1120,196,15077.586207,2412.413793
189,ASUS,Intel,Celeron Dual,Not Available,4 GB,LPDDR4,0 GB,512 GB,Windows,32-bit,...,Casual,No warranty,No,No,17990,3 stars,1180,148,15508.620690,2481.379310
188,ASUS,Intel,Celeron Dual,Not Available,4 GB,LPDDR4,0 GB,512 GB,Windows,64-bit,...,Casual,No warranty,No,No,18990,3 stars,197,25,16370.689655,2619.310345
170,ASUS,Intel,Celeron Dual,Not Available,4 GB,LPDDR4,0 GB,512 GB,Windows,32-bit,...,Casual,No warranty,No,No,20490,3 stars,637,91,17663.793103,2826.206897
163,ASUS,Intel,Celeron Dual,Not Available,4 GB,LPDDR4,0 GB,512 GB,Windows,32-bit,...,Casual,No warranty,No,No,21490,3 stars,818,113,18525.862069,2964.137931
6,ASUS,Intel,Celeron Dual,Not Available,4 GB,DDR4,0 GB,512 GB,Windows,32-bit,...,Casual,No warranty,No,No,21990,3 stars,31,3,18956.896552,3033.103448
5,ASUS,Intel,Celeron Dual,Not Available,4 GB,DDR4,0 GB,512 GB,Windows,64-bit,...,Casual,No warranty,No,No,22990,3 stars,0,0,19818.965517,3171.034483
13,ASUS,Intel,Celeron Dual,Not Available,4 GB,DDR4,0 GB,512 GB,Windows,64-bit,...,Casual,No warranty,No,No,22990,3 stars,0,0,19818.965517,3171.034483
84,ASUS,Intel,Celeron Dual,Not Available,4 GB,LPDDR4,0 GB,512 GB,Windows,32-bit,...,Casual,No warranty,No,No,23490,4 stars,632,92,20250.000000,3240.000000


In [84]:
datos.sort_values(by="Price", ascending=False)  # Ordena por precio en sentido decreciente 

,brand,processor_brand,processor_name,processor_gnrtn,ram_gb,ram_type,ssd,hdd,os,os_bit,...,weight,warranty,Touchscreen,msoffice,Price,rating,Number of Ratings,Number of Reviews,Price before taxes,Taxes
594,ASUS,Intel,Core i9,11th,16 GB,DDR4,3072 GB,0 GB,Windows,32-bit,...,Casual,1 year,No,No,441990,5 stars,1,1,381025.862069,60964.137931
621,APPLE,M1,M1,10th,32 GB,DDR4,1024 GB,0 GB,Mac,64-bit,...,Casual,1 year,No,No,309990,5 stars,3,1,267232.758621,42757.241379
590,ASUS,Intel,Core i9,12th,16 GB,DDR5,1024 GB,0 GB,Windows,64-bit,...,Casual,No warranty,No,No,299990,3 stars,0,0,258612.068966,41377.931034
433,MSI,Intel,Core i5,10th,8 GB,LPDDR4,0 GB,512 GB,DOS,32-bit,...,Casual,No warranty,No,No,291190,4 stars,7,3,251025.862069,40164.137931
547,MSI,Intel,Core i7,12th,16 GB,DDR5,1024 GB,0 GB,Windows,64-bit,...,Casual,No warranty,No,No,279990,3 stars,0,0,241370.689655,38619.310345
591,ASUS,Intel,Core i9,10th,16 GB,DDR4,1024 GB,0 GB,Windows,64-bit,...,Casual,1 year,Yes,No,269990,4 stars,8,1,232750.000000,37240.000000
593,Lenovo,Intel,Core i9,10th,16 GB,DDR4,1024 GB,0 GB,Windows,32-bit,...,Casual,No warranty,No,No,265998,3 stars,0,0,229308.620690,36689.379310
523,Lenovo,Intel,Core i7,8th,16 GB,DDR4,512 GB,0 GB,Windows,64-bit,...,ThinNlight,No warranty,No,No,244801,3 stars,0,0,211035.344828,33765.655172
620,APPLE,M1,M1,10th,16 GB,DDR4,1024 GB,0 GB,Mac,64-bit,...,Casual,1 year,No,No,243990,4 stars,42,8,210336.206897,33653.793103
507,HP,Intel,Core i7,10th,16 GB,LPDDR3,512 GB,0 GB,Windows,64-bit,...,Casual,1 year,Yes,Yes,239759,3 stars,0,0,206688.793103,33070.206897


In [86]:
# Para ordenar el DataFrame utilizando varios criterios
# Ordenar por marca y luego por precio

datos.sort_values(by=["brand", "rating","Price"])


,brand,processor_brand,processor_name,processor_gnrtn,ram_gb,ram_type,ssd,hdd,os,os_bit,...,weight,warranty,Touchscreen,msoffice,Price,rating,Number of Ratings,Number of Reviews,Price before taxes,Taxes
599,APPLE,M1,M1,10th,8 GB,DDR4,256 GB,0 GB,Mac,64-bit,...,Casual,1 year,No,No,85990,4 stars,6725,629,74129.310345,11860.689655
605,APPLE,M1,M1,10th,8 GB,DDR4,256 GB,0 GB,Mac,64-bit,...,Casual,1 year,No,No,85990,4 stars,6725,629,74129.310345,11860.689655
609,APPLE,M1,M1,10th,8 GB,DDR4,512 GB,0 GB,Mac,64-bit,...,Casual,1 year,No,No,101990,4 stars,809,91,87922.413793,14067.586207
608,APPLE,M1,M1,10th,8 GB,DDR4,512 GB,0 GB,Mac,64-bit,...,Casual,1 year,No,No,103990,4 stars,809,91,89646.551724,14343.448276
601,APPLE,M1,M1,10th,16 GB,DDR4,256 GB,0 GB,Mac,64-bit,...,Casual,1 year,No,No,104990,4 stars,45,4,90508.620690,14481.379310
604,APPLE,M1,M1,10th,16 GB,DDR4,256 GB,0 GB,Mac,64-bit,...,Casual,1 year,No,No,104990,4 stars,45,4,90508.620690,14481.379310
610,APPLE,M1,M1,10th,8 GB,DDR4,512 GB,0 GB,Mac,64-bit,...,Casual,1 year,No,No,104990,4 stars,6725,629,90508.620690,14481.379310
616,APPLE,M1,M1,10th,8 GB,DDR4,512 GB,0 GB,Mac,64-bit,...,Casual,1 year,No,No,104990,4 stars,6725,629,90508.620690,14481.379310
600,APPLE,M1,M1,10th,8 GB,DDR4,256 GB,0 GB,Mac,64-bit,...,Casual,1 year,No,No,112990,4 stars,640,70,97405.172414,15584.827586
602,APPLE,M1,M1,10th,8 GB,DDR4,256 GB,0 GB,Mac,64-bit,...,Casual,1 year,No,No,112990,4 stars,640,70,97405.172414,15584.827586


In [89]:
datos  # Observamos que la ordenación del dataframe sigue siendo la misma, es decir
       # nos mostró una copia del dataframe ordenado.

,brand,processor_brand,processor_name,processor_gnrtn,ram_gb,ram_type,ssd,hdd,os,os_bit,...,weight,warranty,Touchscreen,msoffice,Price,rating,Number of Ratings,Number of Reviews,Price before taxes,Taxes
0,ASUS,Intel,Core i3,10th,4 GB,DDR4,0 GB,1024 GB,Windows,64-bit,...,Casual,No warranty,No,No,34649,2 stars,3,0,29869.827586,4779.172414
1,Lenovo,Intel,Core i3,10th,4 GB,DDR4,0 GB,1024 GB,Windows,64-bit,...,Casual,No warranty,No,No,38999,3 stars,65,5,33619.827586,5379.172414
2,Lenovo,Intel,Core i3,10th,4 GB,DDR4,0 GB,1024 GB,Windows,64-bit,...,Casual,No warranty,No,No,39999,3 stars,8,1,34481.896552,5517.103448
3,ASUS,Intel,Core i5,10th,8 GB,DDR4,512 GB,0 GB,Windows,32-bit,...,Casual,No warranty,No,No,69990,3 stars,0,0,60336.206897,9653.793103
4,ASUS,Intel,Celeron Dual,Not Available,4 GB,DDR4,0 GB,512 GB,Windows,64-bit,...,Casual,No warranty,No,No,26990,3 stars,0,0,23267.241379,3722.758621
5,ASUS,Intel,Celeron Dual,Not Available,4 GB,DDR4,0 GB,512 GB,Windows,64-bit,...,Casual,No warranty,No,No,22990,3 stars,0,0,19818.965517,3171.034483
6,ASUS,Intel,Celeron Dual,Not Available,4 GB,DDR4,0 GB,512 GB,Windows,32-bit,...,Casual,No warranty,No,No,21990,3 stars,31,3,18956.896552,3033.103448
7,ASUS,Intel,Core i5,10th,8 GB,DDR4,0 GB,1024 GB,Windows,32-bit,...,Casual,No warranty,No,No,58799,3 stars,0,0,50688.793103,8110.206897
8,Lenovo,Intel,Core i5,10th,4 GB,DDR4,0 GB,1024 GB,Windows,32-bit,...,Casual,No warranty,No,No,49999,3 stars,0,0,43102.586207,6896.413793
9,acer,AMD,Ryzen 5,10th,4 GB,DDR4,0 GB,512 GB,Windows,32-bit,...,Casual,No warranty,No,No,59990,4 stars,1946,240,51715.517241,8274.482759


In [90]:
# Para que el orden se aplique al dataframe original
datos.sort_values(by=["brand", "rating","Price"], inplace=True)

In [91]:
datos  #ahora si ya el DataFrame está ordenado

,brand,processor_brand,processor_name,processor_gnrtn,ram_gb,ram_type,ssd,hdd,os,os_bit,...,weight,warranty,Touchscreen,msoffice,Price,rating,Number of Ratings,Number of Reviews,Price before taxes,Taxes
599,APPLE,M1,M1,10th,8 GB,DDR4,256 GB,0 GB,Mac,64-bit,...,Casual,1 year,No,No,85990,4 stars,6725,629,74129.310345,11860.689655
605,APPLE,M1,M1,10th,8 GB,DDR4,256 GB,0 GB,Mac,64-bit,...,Casual,1 year,No,No,85990,4 stars,6725,629,74129.310345,11860.689655
609,APPLE,M1,M1,10th,8 GB,DDR4,512 GB,0 GB,Mac,64-bit,...,Casual,1 year,No,No,101990,4 stars,809,91,87922.413793,14067.586207
608,APPLE,M1,M1,10th,8 GB,DDR4,512 GB,0 GB,Mac,64-bit,...,Casual,1 year,No,No,103990,4 stars,809,91,89646.551724,14343.448276
601,APPLE,M1,M1,10th,16 GB,DDR4,256 GB,0 GB,Mac,64-bit,...,Casual,1 year,No,No,104990,4 stars,45,4,90508.620690,14481.379310
604,APPLE,M1,M1,10th,16 GB,DDR4,256 GB,0 GB,Mac,64-bit,...,Casual,1 year,No,No,104990,4 stars,45,4,90508.620690,14481.379310
610,APPLE,M1,M1,10th,8 GB,DDR4,512 GB,0 GB,Mac,64-bit,...,Casual,1 year,No,No,104990,4 stars,6725,629,90508.620690,14481.379310
616,APPLE,M1,M1,10th,8 GB,DDR4,512 GB,0 GB,Mac,64-bit,...,Casual,1 year,No,No,104990,4 stars,6725,629,90508.620690,14481.379310
600,APPLE,M1,M1,10th,8 GB,DDR4,256 GB,0 GB,Mac,64-bit,...,Casual,1 year,No,No,112990,4 stars,640,70,97405.172414,15584.827586
602,APPLE,M1,M1,10th,8 GB,DDR4,256 GB,0 GB,Mac,64-bit,...,Casual,1 year,No,No,112990,4 stars,640,70,97405.172414,15584.827586
